core/src/test/java/bio/terra/pearl/core/service/notification/NotificationServiceTest.java

```diff
@@ -73,4 +77,53 @@ void findAllByConfigIdWithEnrollees(TestInfo info) {
         Assertions.assertEquals(bundle1.enrollee(), foundNotif1.getEnrollee());
         Assertions.assertEquals(bundle2.enrollee(), foundNotif2.getEnrollee());
     }
+
+    @Test
+    @Transactional
+    void findAllByConfigIdAttachSendgridEvents(TestInfo info) {
+
+        EnrolleeFactory.EnrolleeBundle bundle1 = enrolleeFactory.buildWithPortalUser(getTestName(info));
+        EnrolleeFactory.EnrolleeBundle bundle2 = enrolleeFactory.buildWithPortalUser(getTestName(info));
+        EnrolleeFactory.EnrolleeBundle bundle3 = enrolleeFactory.buildWithPortalUser(getTestName(info));
+
+        Trigger trigger1 = triggerFactory.buildPersisted(getTestName(info));
+
+        Notification notif1 = notificationFactory.buildPersisted(bundle1, trigger1);
+        Notification notif2 = notificationFactory.buildPersisted(bundle2, trigger1);
+        Notification notif3 = notificationFactory.buildPersisted(bundle2, trigger1);
+        Notification notif4 = notificationFactory.buildPersisted(bundle3, trigger1);
+
+        sendgridEventDao.create(SendgridEvent.builder()
+                .notificationId(notif1.getId())
+                .status("delivered")
+                .msgId("mock-id-1")
+                .build());
+
+        sendgridEventDao.create(SendgridEvent.builder()
+                .notificationId(notif4.getId())
+                .status("opened")
+                .msgId("mock-id-2")
+                .build());
+
+        sendgridEventDao.create(SendgridEvent.builder()
+                .notificationId(notif3.getId())
+                .status("n/a")
+                .msgId("mock-id-3")
+                .build());
+
+        List<Notification> notifs = notificationService.findAllByConfigId(trigger1.getId(), true);
+
+        Assertions.assertEquals(4, notifs.size());
+
+        Notification foundNotif1 = notifs.stream().filter(notif -> notif.getId().equals(notif1.getId())).findFirst().orElseThrow();
+        Notification foundNotif2 = notifs.stream().filter(notif -> notif.getId().equals(notif2.getId())).findFirst().orElseThrow();
+        Notification foundNotif3 = notifs.stream().filter(notif -> notif.getId().equals(notif3.getId())).findFirst().orElseThrow();
+        Notification foundNotif4 = notifs.stream().filter(notif -> notif.getId().equals(notif4.getId())).findFirst().orElseThrow();
+
+
+        Assertions.assertEquals(foundNotif1.getEventDetails().getStatus(), "delivered");
+        Assertions.assertNull(foundNotif2.getEventDetails());
+        Assertions.assertEquals(foundNotif3.getEventDetails().getStatus(), "n/a");
+        Assertions.assertEquals(foundNotif4.getEventDetails().getStatus(), "opened");
+    }
```

Comment:
nit: check that your intellij is set to always add newlines at the end of files on save